tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -332,7 +374,104 @@ TEST_F(OverlapTest, ReduceScatterBasedPipeliningHostIrImplementation) {
 
     hie.runWithInput(std::move(inputs));
   }
-  validate();
+}
+
+class RingBasedOverlapTest : public OverlapTest {
+ protected:
+  at::Tensor src_buffer_, dst_buffer_;
+  void SetUp() override {
+    OverlapTest::SetUp();
+
+    std::vector<int64_t> buffer_sizes = {
+        params.S, params.M / params.S, params.N};
+    src_buffer_ = at::empty(buffer_sizes, gpu_options_);
```

Comment:
further optimizations are possible here to reduce the footprint of the src and dst buffers:
- One slice can be shared between src and dst, corresponding to the "local" send/recv at the last iteration
- Probably possible to reduce the outermost dimension with the number of streams.

Imo it would be reasonable to keep those optimizations for later when we have proper perf analysis